controller/src/main/java/org/jboss/as/controller/PathAddress.java

```diff
@@ -451,6 +451,33 @@ public String toPathStyleString() {
         return toString('/');
     }
 
+    /**
+     * Check if this path matches the address path.
```

Comment:
This could also use a bit of javadoc mentioning how it differs from equals by checking if the elements match rather than if they are equal.